src/peer_manager.rs

```diff
@@ -353,23 +343,14 @@ impl PeerManager {
         );
     }
 
-    /// Adds a potential candidate to the candidate list setting its state to `VotedFor`.  If
-    /// another ongoing (i.e. unapproved) candidate exists, or if the candidate is unsuitable for
-    /// adding to our section, returns an error.
-    pub fn expect_candidate(&mut self, old_pub_id: PublicId) -> Result<(), RoutingError> {
-        if self.candidate != Candidate::None {
-            return Err(RoutingError::AlreadyHandlingJoinRequest);
-        }
-        self.candidate = Candidate::Expecting {
-            timestamp: Instant::now(),
-            old_pub_id: old_pub_id,
-        };
-        Ok(())
+    /// Return true if already has a candidate
+    pub fn has_candidate(&self) -> bool {
+        self.candidate != Candidate::None
```

Comment:
I was going to recommend using `is_some` before I realised this `None` was a `Candidate` variant.

Side-note: should we maybe use an Option of Candidate in many of these places rather than mixing concepts in Candidate? (I'm not expecting you to do it here since it's pre-existing code... Just throwing a thought out there)

Reply:
I am not sure what is the best pattern, both express the same exactly the same thing. We should have this discussion at some point.

Reply:
Sure. We can chat about it. My argument for using `Option` is that you get the nicities like `is_some`, `map`, `and_then`,... for free without having to manually re-code them everywhere like a barbarian :wink:

Reply:
Also, in terms of SRP, if you're an `Option`, you're an `Option`. If you're a `Candidate`, you're a `Candidate`... It's not nearly as bad as mixing memory management with other functionality, but it's got similarities with that anti-pattern...

Reply:
Moved that conversation to rust-help